### PR TITLE
added cfd metadata and keys

### PR DESCRIPTION
**Pull request by khiltunen**
This PR adds missing CFD engines metadata and keys

### CODE REVIEW

**Comment by khiltunen** [comment]
@stefanoborini and @ahashibon could you review the added CFD metadata and keys

File: yaml_files/cuba.yml [hunk 1]
```diff
@@ -516,3 +516,22 @@ CUBA_KEYS:
...
     definition: A lattice parameter to define a Bravais Lattice
     type: double
+
+  STEADY_STATE:
+    definition: Use steady state solver
+    type: boolean
+
+  MAXIMUM_COURANT_NUMBER:
+    definition: Maximum Courant number
+    shape: [1]
```
Comment (stefanoborini) [inline yaml_files/cuba.yml]: You don't need the shape, as the default is [1]

File: yaml_files/cuba.yml [hunk 2]
```diff
@@ -516,3 +516,22 @@ CUBA_KEYS:
...
+    type: double
+
+  NUMBER_OF_CORES:
+    definition: Number of cpu cores to use
+    shape: [1]
```
Comment (stefanoborini) [inline yaml_files/cuba.yml]: same

File: yaml_files/cuba.yml [hunk 3]
```diff
@@ -516,3 +516,22 @@ CUBA_KEYS:
...
+    type: integer
+
+  MAGNITUDE:
+    definition: Magnitude of vector variable
+    shape: [1]
```
Comment (stefanoborini) [inline yaml_files/cuba.yml]: same

File: yaml_files/simphony_metadata.yml [hunk 4]
```diff
@@ -266,6 +266,19 @@ CUDS_KEYS:
     CUBA.POWER_LAW_INDEX:
       default: 0.5
 
+  BINGHAM_PLASTIC_MODEL:
+    definition: Bingham plastic viscosity model (only in mixture model)
+    parent: CUBA.RHEOLOGY_MODEL
+    models: [CUBA.CONTINUUM]
+    CUBA.LINEAR_CONSTANT:
+      shape: (2)
```
Comment (stefanoborini) [inline yaml_files/simphony_metadata.yml]: I would use [] instead. () is valid as well, but we use it only when we have a : entry

File: yaml_files/simphony_metadata.yml [hunk 5]
```diff
@@ -266,6 +266,19 @@ CUDS_KEYS:
...
+      default: [0.0, 0.0]
+    CUBA.POWER_LAW_INDEX:
+      shape: (2)
```
Comment (stefanoborini) [inline yaml_files/simphony_metadata.yml]: same

File: yaml_files/simphony_metadata.yml [hunk 6]
```diff
@@ -439,6 +487,49 @@ CUDS_KEYS:
...
+    CUBA.VARIABLE:
+      shape: (:)
+      default: []
+    CUBA.MATERIAL:
+      shape: (:)
+      default: []
+
+  WETTING_ANGLE:
+    parent: CUBA.CONDITION
+    definition: Volume fraction wall boundary condition with specified contatc angle
```
Comment (stefanoborini) [inline yaml_files/simphony_metadata.yml]: contact

**Review by stefanoborini — CHANGES_REQUESTED** [review]

**Comment by stefanoborini** [comment]
reviewed for file format. science for @ahashibon

**Pull request merged**